docs/maps/terms-join.asciidoc

```diff
@@ -0,0 +1,119 @@
+[[terms-join]]
+=== Terms join
+
+*Terms joins* combine the result of an Elasticsearch terms aggregation and vector features based on a shared key.
+
+Use *Terms joins* to augment vector features with additional properties.
+You can then use these additional property values to symbolize features
+and provide richer tooltip context.
+
+[role="screenshot"]
+image::maps/images/terms_join.png[]
+
+Follow the example below to understand how *Terms joins* augments vector features with additional properties.
+This example will use Elastic Maps Service(EMS) World Countries as the vector source and
+the Kibana sample data set "Sample web logs" as the elasticsearch index.
+
+Example feature from World Countries
+--------------------------------------------------
+{
+  geometry: {
+    coordinates: [...],
+    type: "Polygon"
+  },
+  properties: {
+    name: "Sweden",
+    iso2: "SE",
+    iso3: "SWE"
+  },
+  type: "Feature"
+}
+--------------------------------------------------
+
+Example documents from Sample web logs
+--------------------------------------------------
+{
+  bytes: 1837,
+  geo: {
+    src: "SE"
+  },
+  timestamp: "Feb 28, 2019 @ 07:23:08.754"
+},
+{
+  bytes: 971,
+  geo: {
+    src: "SE"
+  },
+  timestamp: "Feb 27, 2019 @ 08:10:45.205"
+},
+{
+  bytes: 4277,
+  geo: {
+    src: "SE"
+  },
+  timestamp: "Feb 21, 2019 @ 05:24:33.945"
+},
+{
+  bytes: 5624,
+  geo: {
+    src: "SE"
+  },
+  timestamp: "Feb 21, 2019 @ 04:57:05.921"
+}
+--------------------------------------------------
+
+The JOIN configuration links the vector source "World Countries" to the elasticsearch index "kibana_sample_data_logs"
+on the shared key *iso2 = geo.src*.
+[role="screenshot"]
+image::maps/images/terms_join_shared_key_config.png[]
+
+The METRICS configuration defines two metric aggregations;
+the count of all documents in the terms bucket and
+the average of the field "bytes" for all documents in the terms bucket.
+[role="screenshot"]
+image::maps/images/terms_join_metric_config.png[]
+
+Example terms aggregation response
+--------------------------------------------------
+{
+  aggregations: {
+    join: {
+      buckets: [
+        {
+          doc_count: 4,
+          key: "SE",
+          avg_of_bytes: {
+            value: 3177.25
+          }
+        }
+      ]
+    }
+  }
+}
+--------------------------------------------------
+
+Finally, the terms aggregation response is then joined with the vector features.
+--------------------------------------------------
+{
+  geometry: {
+    coordinates: [...],
+    type: "Polygon"
+  },
+  properties: {
+    name: "Sweden",
+    iso2: "SE",
+    iso3: "SWE",
+    join_prop_count: 4,
+    join_prop_avg_of_bytes: 3177.25
+  },
+  type: "Feature"
+}
+--------------------------------------------------
+
+NOTE: This example is for demonstration purposes only and the actual implemention uses different key
+names to avoid naming collisions with vector feauture properties.
```

Comment:
to avoid having to include this hedge, we could just show a picture of a tooltip showing all these fields.

showing the json-doc isn't really relevant to the user either imho.